openforcefield/tests/test_forcefield.py

```diff
@@ -1557,6 +1594,437 @@ def test_lookup_parameter_handler_object(self):
+                match="once" >
+            </VirtualSite>
+        </VirtualSites>
+    </SMIRNOFF>
+    """
+
+    xml_ff_virtual_sites_trivalent_match_all = """
+    <?xml version="1.0" encoding="ASCII"?>
+    <SMIRNOFF version="0.3" aromaticity_model="OEAroModel_MDL">
+            <VirtualSite
+                type="TrivalentLonePair"
+                name="EP"
+                smirks="[*:1]-[#7X3:2]-([*:3])-[*:4]"
+                distance="0.70 * angstrom"
+                chargeincrement1="0.1*elementary_charge"
+                chargeincrement2="0.1*elementary_charge"
+                chargeincrement3="0.1*elementary_charge"
+                chargeincrement4="0.1*elementary_charge"
+                sigma="0.1*angstrom"
+                epsilon="0.1*kilocalories_per_mole"
+                match="once" >
+            </VirtualSite>
+        </VirtualSites>
+    </SMIRNOFF>
+    """
+
+    def _test_physical_parameters(self, tkr, xml, smi, assert_physics, mol=None):
+
+        from simtk.openmm import NonbondedForce
+
+        forcefield = ForceField("openff-1.2.0.offxml", xml)
+        if mol is None:
+            mol = Molecule.from_smiles(smi)
+        topology = mol.to_topology()
+        kwargs = {}
+        if mol.partial_charges is not None:
+            kwargs["charge_from_molecules"] = [mol]
+        omm_system = forcefield.create_openmm_system(
+            topology, toolkit_registry=tkr, **kwargs
+        )
+        nonbondedForce = [
+            f for f in omm_system.getForces() if type(f) == NonbondedForce
+        ][0]
+
+        for particle_index, expected in enumerate(assert_physics):
+            parameters = nonbondedForce.getParticleParameters(particle_index)
+            assert all(
+                [
+                    (b is None or np.isclose(a / a.unit, b / a.unit))
+                    for a, b in zip(parameters, expected)
+                ]
+            )
+        return
+
+    q_unit = unit.elementary_charge
+    e_unit = unit.kilocalorie_per_mole
+    s_unit = unit.angstrom
+
+    ############################################################################
+    # Bond charge virtual site test data
+    ############################################################################
+
+    # First test that one vsite is added, and that the last parameter
+    # is chosen over the top-most wildcard match
+    opts = OrderedDict(
+        {
+            "xml": xml_ff_virtual_sites_bondcharge_match_once,
+            "smi": "[N]#[N]",
+            "assert_physics": (
+                # charge        sigma         epsilon
+                (+0.2 * q_unit, None, None),
+                (+0.2 * q_unit, None, None),
+                (-0.4 * q_unit, 0.2 * s_unit, 0.2 * e_unit),
+            ),
+        }
+    )
+    bond_charge_parameters_args.append(opts)
+
+    # Test the wildcard match
+    opts = OrderedDict(
+        {
+            "xml": xml_ff_virtual_sites_bondcharge_match_once,
+            "smi": "[C][C]",
```

Comment:
Since atom indexing of molecules from SMILES are arbitrary, and here we're checking atom indices in the final system, it could be good to make a `create_ethane` method (like `create_ethanol`) to seed this test.